code/mccSearch.py

```diff
@@ -542,7 +612,838 @@ def find_cloud_elements(mergImgs, timelist, mainStrDir, lat, lon, TRMMdirName=No
+            #populate the dictionary
+            if TRMMdirName:
+                cloudElementDict = {'uniqueID': ceUniqueID, 'cloudElementTime': timelist[t],\
+                    'cloudElementLatLon': cloudElementLatLons, 'cloudElementCenter':cloudElementCenter, \
+                    'cloudElementArea':cloudElementArea, 'cloudElementEccentricity':cloudElementEpsilon, \
+                    'cloudElementTmax':TIR_max, 'cloudElementTmin': TIR_min, 'cloudElementPrecipTotal':precipTotal,\
+                    'cloudElementLatLonTRMM':ceTRMMList, 'TRMMArea': TRMMArea, 'CETRMMmax':maxCEprecipRate, \
+                    'CETRMMmin':minCEprecipRate}
+            else:
+                cloudElementDict = {'uniqueID': ceUniqueID, 'cloudElementTime': timelist[t],\
+                    'cloudElementLatLon': cloudElementLatLons, 'cloudElementCenter':cloudElementCenter, \
+                    'cloudElementArea':cloudElementArea, 'cloudElementEccentricity':cloudElementEpsilon, \
+                    'cloudElementTmax':TIR_max, 'cloudElementTmin': TIR_min,}
+   
+            #Store the new cloudElementDict so that it can be returned to parent function
+            allCloudElementDicts.append(cloudElementDict)
+            
+            #Will hold data for JSON
+            currentJSONDict = {}
+            
+            # get some data for the JSON object which will only store the graph CEs and connected edges
+            currentJSONDict['cf'] = (((ndimage.minimum(cloudElement, \
+                labels=labels)) / float((ndimage.maximum(cloudElement, labels=labels)))) * 100.0)
+            currentJSONDict['tmin'] = ndimage.minimum(cloudElement, labels=labels)*1.
+            currentJSONDict['tmax'] = ndimage.maximum(cloudElement, labels=labels)*1.
+            currentJSONDict['time'] = str(timelist[t])
+            currentJSONDict['area'] = cloudElementArea
+            currentJSONDict['center'] = cloudElementCenter
+            currentJSONDict['lat_lon_box'] = latLonBox
+            currentJSONDict['shape'] = shape
+            currentJSONDict['cloudElementEpsilon'] = cloudElementEpsilon
+            
+            allJSONDataDicts.append(currentJSONDict)
+            
+            #cf = (((ndimage.minimum(cloudElement, \
+            #    labels=labels)) / float((ndimage.maximum(cloudElement, labels=labels)))) * 100.0)
+            #tmin = ndimage.minimum(cloudElement, labels=labels)*1.
+            #tmax = ndimage.maximum(cloudElement, labels=labels)*1.
+            #if edges:
+            #    cloudElementsJSON.append({'cloudElement': ceUniqueID, 'time': str(timelist[t]),\
+            #        'area':cloudElementArea, 'Tmax': tmax, 'Tmin': tmin,'center':cloudElementCenter,\
+            #        'convective_fraction': cf, 'lat_lon_box': latLonBox, 'shape': shape, 'edges':edges, 'eccentricity':cloudElementEpsilon})
+
+            profTimes[8]+= time.time() - pin
+            pin = time.time() #tenth pin drop
+
+            if frameNum==1:
```

Comment:
This if statement is incorrect. I believe you mis-indented from the original code. According to the original code, this else statement was not for the "if frameNum != 1", but for the if statement "if cloudElementArea >= AREA_MIN or (cloudElementArea < AREA_MIN and \
                    ((ndimage.minimum(cloudElement, labels=labels)) / \
                        float((ndimage.maximum(cloudElement, labels=labels)))) < CONVECTIVE_FRACTION):"